vmmlib/vector.hpp

```diff
@@ -324,57 +303,19 @@ template< size_t M, typename T > class vector
         return os;
     }
 
-    T array[ M ];    //!< storage
-
-#ifndef SWIG
-    // Vector3 defaults
-    static const vector FORWARD;
-    static const vector BACKWARD;
-    static const vector UP;
-    static const vector DOWN;
-    static const vector LEFT;
-    static const vector RIGHT;
-
-    static const vector ONE;
-    static const vector ZERO;
-
-    // Unit vectors
-    static const vector UNIT_X;
-    static const vector UNIT_Y;
-    static const vector UNIT_Z;
-#endif
-
-}; // class vector
+    /** @name Convenience presets for 3 and 4 component vectors */
+    //@{
+    static vector< M, T > zero();
```

Comment:
I am not sure what to think about these functions replacing the constants. I somehow liked them more, maybe because I was used to them from Ogre3d:
http://www.ogre3d.org/docs/api/1.9/class_ogre_1_1_vector3.html

In non-templated code such factory functions (for matrices at least) make sense but are often called zeros(n,m) and ones(n,m) using the plural (e.g. Numpy, OpenCV, etc..).

In GLM there does not seem to be either of the two, however I discovered that they have constructors which take a (vector of size N-1, scalar) for the homogeneous coordinates:
glm::vec4 Position = glm::vec4(glm::vec3(0.0), 1.0); // https://glm.g-truc.net/0.9.2/api/a00001.html

I am also mildly convinced about the forward() / backward() / etc. because they are only valid for a certain coordinates system. Having UNIT_X, NEGATIVE_UNIT_Y, etc. like in Ogre look more correct to me.